service/tablet_allocator.cc

```diff
@@ -638,6 +649,131 @@ class load_balancer {
         return utils::get_local_injector().enter("tablet_allocator_shuffle");
     }
 
+    shard_id rand_shard() const {
+        static thread_local std::default_random_engine re{std::random_device{}()};
+        static thread_local std::uniform_int_distribution<shard_id> dist;
+        return dist(re);
```

Comment:
This isn't a rand_shard. Better pass the shard count and mod it here.

Reply:
Right